### ISSUE TITLE
PopupDemo regression

### ISSUE THREAD

**Issue by TomasMikula** [TomasMikula/RichTextFX#220]
There is a regression in the PopupDemo (`gradle PopupDemo`), introduced after 0.6.10: the position of the popup does not update as I type.

**Comment by JordanMartinez** [comment]
The regression doesn't exist as of 7277f9f8a99b6a4933e344a13fefcd474a041aac (the last commit before the skin was removed), but shows up somewhere after I removed the skin. When my work was merged in 45145ff2f409d4ede84fa04eacb2540ce84e40eb, this regression appears.

I've narrowed it down to this. When user types/deletes a character, the following path occurs:
````java
layoutChildren() // calls positionPopup()
---
positionPopup() { // calls getSelectionBoundsOnScreen()
switch(alignment.getAnchorObject()) {
        case CARET:
            bounds = getCaretBoundsOnScreen(); break;
        case SELECTION:
            bounds = getSelectionBoundsOnScreen(); break;
    }
    // at this point, bounds is null when it should have a value to which
    // popup would position itself.
    bounds.ifPresent(b -> { /* code */ })
}
---
getSelectionBoundsOnScreen() { 
// calls getCaretBoundsOnScreen 
// since selection's length is 0
    IndexRange selection = getSelection();
        if(selection.getLength() == 0) {
            return getCaretBoundsOnScreen();
        }
}
---
getCaretBoundsOnScreen() {
    // virtualFlow.getCellIfVisible(getCurrentParagraph()).isPresent() == false
    return virtualFlow.getCellIfVisible(getCurrentParagraph())
        .map(c -> c.getNode().getCaretBoundsOnScreen());
}
````

**Comment by JordanMartinez** [comment]
Tomas, when a user selects text from the left and going right in the demo, should the popup be (almost) directly underneath where the caret is as it progresses to the right?

When I positioned the caret somewhere, pressed and held Shift, and used the right arrow to further select characters on the right, the popup didn't line up vertically with the caret.

**Comment by JordanMartinez** [comment]
Changing `getCaretBoundsOnScreen` to the following makes it work. So, is this a regression in `Flowless`?
````java
Cell<Paragraph<S, PS>, ParagraphBox<S, PS>> cell = virtualFlow.getCell(getCurrentParagraph());
    if (virtualFlow.visibleCells().contains(cell)) {
        return Optional.of(cell.getNode().getCaretBoundsOnScreen())
    }
````

**Comment by JordanMartinez** [comment]
The code on the commit immediately after 7277f9f (where Flowless is updated to the then-current 0.5-Snapshot) works, so I guess it really is some thing caused in my code and note Flowless's new version.

**Comment by JordanMartinez** [comment]
When a user types a letter, `positionPopup()` (and therefore `layoutChildren()`) is called twice in the pre-skin-removal commits but is only called once in the post-skin-removal commits.

The necessary bounds that would update the popup position are not present in the first call but are in the second. So, I'm guessing that my code somehow prevents the second call to `layoutChildren()`

**Comment by JordanMartinez** [comment]
Ok, I redid a one of my skin-removal commits (I inlined StyledTextAreaView's content into StyledTextArea) in [my branch "regressionCheck"](https://github.com/JordanMartinez/RichTextFX/tree/checkRegression), along with a few other commits that later followed: popupAnchorAdjustment's naming conflict and removing a reference to view/virtualFlow in `StyledTextAreaBehavior` because it's no longer needed. 

In this branch's second-to-last commit, there is no regression in the code. Despite comparing the differences between `regressionCheck` and the current master branch, I can't find major differences that would show what's causing the problem. 

However, I have found that the regression does appear in `regressionCheck` when I move `VirtualizedScrollPane` outside of `STA`.

So, the last commit in the branch (currently `7c44fa08de37fd1fc7440b64c71bba4fe5795b78`) includes the regression (which can be removed if one follows the commenting out instructions: `// ### comment out instructions ###`

**Comment by JordanMartinez** [comment]
Wrapping `positionPopup()` in `Platform.runLater()` fixes the problem, but I'm not sure if that's the cleanest way to fix this regression. See my PR #238

**Comment by JordanMartinez** [comment]
Turns out this was a deficiency in Flowless. See #238 and TomasMikula/Flowless#19

**Comment by TomasMikula** [comment]
Fixed by TomasMikula/Flowless#20
